daemon/container_unix.go

```diff
@@ -948,7 +948,8 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 
 	ep, err := container.getEndpointInNetwork(n)
 	if err == nil {
-		return fmt.Errorf("container already connected to network %s", idOrName)
```

Comment:
If it is trying to address it from a UX perspective, we can do something different.
`connectToNetwork` is called from `network connect` and `docker run`. So, I would avoid adding this logic to this function.
The UX counterpart for this is `ConnectToNetwork`. Please note the subtle :) exported function which is used only for `docker network connect` and we could add some additional intelligence and also take multi-host into consideration. But I think this is not critical for 1.9.0